src/components/button/Button.props.ts

```diff
@@ -9,10 +10,10 @@ import {
     TButtonWidth,
 } from './Button.types';
 
-type PButton = {
+type PButton = Partial<PAriaAllAttributes> & {
```

Comment:
Not all aria attributes are appropriate everywhere. Should we be move specific on a component-by-component basis with what aria attributes are available? I get that is more work for us, but doing so would help ensure that those not as familiar with aria will be able to more readily tell what attributes a component can actually use.

Regardless, we should be sure to include guidelines in storybook for properly implementing accessibility when using Compass Components.

Reply:
we definitely should ... BUT ... it is a lot to define all of it correctly ... aria-attributes are selected by either role, HTML element or attributes, so this is no easy task.

I added the complete Aria type for now to test the type "correctness" and functionality.

Correctly typing out all roles would be good starting point IMO. There is a good list (with corresonding aria-attributes that come with the role) to be found [here](https://www.w3.org/TR/wai-aria-1.0/roles#role_definitions)

Reply:
Okay, sounds good. Can you please create Jira tickets for the next steps for Aria support so we can keep track of them following merging this initial implementation?

Reply:
I did some research ... IMO it is not a good investment to add the complete generic aria-functionality like we discussed earlier, since ...
a) JSX already supports all aria-attributes out of the box and ...
b) the effort to implement it based on roling (which is not always clear, or influenced by how the implementation is done) is way too high.

I left a comment/thread in MM so we can have a more in-depth discussion there.

For now I removed the Aria Type and the computed setting of 'aria-label' from the `Button` props